utility_network/perform-valve-isolation-trace/src/main/java/com/esri/samples/perform_valve_isolation_trace/PerformValveIsolationTraceController.java

```diff
@@ -363,68 +341,74 @@ private void handleMapViewClicked(MouseEvent e) {
+              if (utilityElement.getNetworkSource().getSourceType() == UtilityNetworkSource.Type.JUNCTION) {
 
-                  // apply the selected terminal
-                  if (userSelectedTerminal.isPresent()) {
-                    UtilityTerminal terminal = userSelectedTerminal.get();
-                    utilityElement.setTerminal(terminal);
-                    // show the terminals name in the status label
-                    String terminalName = terminal.getName() != null ? terminal.getName() : "default";
-                    statusLabel.setText("Feature added at terminal: " + terminalName);
+                // check if the feature has a terminal configuration and multiple terminals
+                if (utilityElement.getAssetType().getTerminalConfiguration() != null) {
+                  UtilityTerminalConfiguration utilityTerminalConfiguration = utilityElement.getAssetType().getTerminalConfiguration();
+                  List<UtilityTerminal> terminals = utilityTerminalConfiguration.getTerminals();
 
-                    // don't create the element if no terminal was selected
-                  } else {
-                    statusLabel.setText("No terminal selected - no feature added");
-                    return;
+                  if (terminals.size() > 1) {
+                    // prompt the user to select a terminal for this feature
+                    Optional<UtilityTerminal> userSelectedTerminal = promptForTerminalSelection(terminals);
+
+                    // apply the selected terminal
+                    if (userSelectedTerminal.isPresent()) {
+                      UtilityTerminal terminal = userSelectedTerminal.get();
+                      utilityElement.setTerminal(terminal);
+                      // show the terminals name in the status label
+                      String terminalName = terminal.getName() != null ? terminal.getName() : "default";
+                      statusLabel.setText("Feature added at terminal: " + terminalName);
+
+                      // don't create the element if no terminal was selected
+                    } else {
+                      statusLabel.setText("No terminal selected - no feature added");
+                    }
                   }
                 }
-              }
 
-            } else if (utilityElement.getNetworkSource().getSourceType() == UtilityNetworkSource.Type.EDGE) {
+              } else if (utilityElement.getNetworkSource().getSourceType() == UtilityNetworkSource.Type.EDGE) {
 
-              // get the geometry of the identified feature as a polyline, and remove the z component
-              Polyline polyline = (Polyline) GeometryEngine.removeZ(identifiedFeature.getGeometry());
+                // get the geometry of the identified feature as a polyline, and remove the z component
+                Polyline polyline = (Polyline) GeometryEngine.removeZ(listOfFeatures.get(0).getGeometry());
```

Comment:
Is it always going to be the first one? `listOfFeatures.get(0)`

Reply:
from the sample design, yeh that's the one we want! thanks for checking :)